tests/test_transformer.py

```diff
@@ -37,6 +40,16 @@ def min_time_limit(self, cqm):
         return 1
 
 
+class MockNL():
+    def sample(self, NL: Model, *, time_limit: float, label: str):
+        sampler = LeapHybridNLSampler()
+        return sampler.sample(NL)
```

Comment:
The issue with the missing API key for just NL is due to CQM using a local CQM solver for testing (thus the mocking). This should be changed to update the NL model _without_ having to call the Leap hybrid NL sampler.

Reply:
Something like this might work, where `FEASIBLE_NL_SOLUTION` would be a known hard-coded solution for a specific input. This input/solution pair can then be used in the tests.
```suggestion
class MockNL():
    def sample(self, NL: Model, *, time_limit: float, label: str) -> None:
        model.states.resize(1)

        for decision in model.iter_decisions():
            decision.set_state(0, FEASIBLE_NL_SOLUTION)
```
I haven't tested this yet, so might need some changes.